shopfloor_reception_packaging_dimension/services/reception.py

```diff
@@ -0,0 +1,237 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo.osv import expression
+
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+from odoo.addons.shopfloor.utils import to_float
+
+
+class Reception(Component):
+    _inherit = "shopfloor.reception"
+
+    @property
+    def dimension_to_update(self):
+        if not hasattr(self.work, "dimension_to_update"):
+            self.work.dimension_to_update = {}
+        return self.work.dimension_to_update
+
+    def _before_response_for_set_quantity(self, picking, line):
```

Comment:
I don't understand why you needed a new method to override instead of overriding `_response_for_set_qty` :thinking:

Reply:
It seemed easier at the time. Looking at it now, because `_response_for_set_quantity` is called in many places in the scenario code and only in a few places we need to inject the packaging dimension check to return a specific response.

Reply:
`_response_for_set_quantity` could be called from elsewhere, it's nice to be able to use this hook or not.
I would check if there's packaging available to set dimension where it's relevant, and return `_response_for_set_packaging_dimension` when we find some.